lib/mayaUsd/ufe/StagesSubject.cpp

```diff
@@ -284,6 +285,25 @@ void attributeMetadataChanged(
         sendAttributeMetadataChanged(ufePath, changedToken, changeType, metadataKeys);
     }
 }
+
+std::vector<std::string> getMetadataKeys(const std::string& strMetadata)
+{
+    std::vector<std::string> metadataKeys;
+    const std::regex         rgx("(')([^ ]*)(':)");
```

Comment:
You are using the power of regular expressions, but not to their full extent. Please read more about capture groups. The function can be rewritten as:
```
std::vector<std::string> getMetadataKeys(const std::string& strMetadata)
{
    std::vector<std::string> metadataKeys;
    const std::regex         rgx("'([^ ]*)':");
    for (std::sregex_iterator it(strMetadata.begin(), strMetadata.end(), rgx), it_end; it != it_end;
         ++it) {
        metadataKeys.push_back(std::string((*it)[1]));
    }
    return metadataKeys;
}
```
And work exactly as requested by extracting the captured strings.